internal/dataplane/translator/ingressrules.go

```diff
@@ -130,6 +105,96 @@ func (ir *ingressRules) populateServices(
 	return serviceNamesToSkip
 }
 
+func (ir *ingressRules) handleServiceClientCertificates(
+	s store.Storer,
+	k8sService *corev1.Service,
+	service *kongstate.Service,
+	failuresCollector *failures.ResourceFailuresCollector,
+) {
+	secretName := annotations.ExtractClientCertificate(k8sService.Annotations)
+	if secretName != "" {
+		secretKey := k8sService.Namespace + "/" + secretName
+		secret, err := s.GetSecret(k8sService.Namespace, secretName)
+		if err != nil {
+			failuresCollector.PushResourceFailure(
+				fmt.Sprintf("Failed to fetch secret '%s': %v", secretKey, err), k8sService,
+			)
+			return
+		}
+
+		// override protocol isn't set yet, need to get it from the annotation
+		protocol := annotations.ExtractProtocolName(k8sService.Annotations)
+		// annotation value does not indicate the effective default, so stuff it in unset
+		if protocol == "" {
+			protocol = "http"
+		}
+		if lo.Contains(getNonTLSProtocols(), protocol) {
+			failuresCollector.PushResourceFailure(
+				fmt.Sprintf("client certificate requested for incompatible service protocol '%s'", *service.Protocol),
+				k8sService,
+			)
+			return
+		}
+		// ensure that the cert is loaded into Kong
+		ir.SecretNameToSNIs.addUniqueParents(secretKey, k8sService)
+		service.ClientCertificate = &kong.Certificate{
+			ID: kong.String(string(secret.UID)),
+		}
+	}
+}
+
+func (ir *ingressRules) handleServiceCACertificates(
+	s store.Storer,
+	service *corev1.Service,
+	k *kongstate.Service,
+	collector *failures.ResourceFailuresCollector,
+) {
+	certificates := annotations.ExtractCACertificates(service.Annotations)
+	if len(certificates) == 0 {
+		// No CA certificates to process.
+		return
+	}
+
+	// Validate that the service has TLS verification turned on.
+	if k.TLSVerify == nil || !*k.TLSVerify {
```

Comment:
The function `handleServiceCACertificates` is called before calling `overrideByAnnotation` on `kongstate.Service` so the `TLSVerify` of the kongstate service is not added from annotations?